pkg/services/hcloud/server/server.go

```diff
@@ -189,6 +189,22 @@ func (s *Service) Reconcile(ctx context.Context) (res reconcile.Result, err erro
 	return res, nil
 }
 
+// implments setting rate limit  on hcloudmachine.
+func handleRateLimit(hm *infrav1.HCloudMachine, err error, functionName string, errMsg string) error {
```

Comment:
@yrs147 please write a unit-test for that function. You can see an example in in server_test.go (`Describe("Test ValidateLabels"`)

Reply:
@guettli I have added the test for `handleRateLimit` method

cc @janiskemper